Source/Scene/PostProcess.js

```diff
@@ -0,0 +1,570 @@
+define([
+        '../Core/BoundingRectangle',
+        '../Core/Check',
+        '../Core/Color',
+        '../Core/combine',
+        '../Core/createGuid',
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/DeveloperError',
+        '../Core/loadImage',
+        '../Core/Math',
+        '../Core/PixelFormat',
+        '../Core/destroyObject',
+        '../Renderer/PassState',
+        '../Renderer/PixelDatatype',
+        '../Renderer/RenderState',
+        '../Renderer/Sampler',
+        '../Renderer/Texture',
+        '../Renderer/TextureMagnificationFilter',
+        '../Renderer/TextureMinificationFilter',
+        '../Renderer/TextureWrap',
+        '../ThirdParty/when',
+        './PostProcessSampleMode'
+    ], function(
+        BoundingRectangle,
+        Check,
+        Color,
+        combine,
+        createGuid,
+        defaultValue,
+        defined,
+        defineProperties,
+        DeveloperError,
+        loadImage,
+        CesiumMath,
+        PixelFormat,
+        destroyObject,
+        PassState,
+        PixelDatatype,
+        RenderState,
+        Sampler,
+        Texture,
+        TextureMagnificationFilter,
+        TextureMinificationFilter,
+        TextureWrap,
+        when,
+        PostProcessSampleMode) {
+    'use strict';
+
+    /**
+     * Runs a post-process on either the texture rendered by the scene or the output of a previous post-process.
+     *
+     * @alias PostProcess
+     * @constructor
+     *
+     * @param {Object} options An object with the following properties:
+     * @param {String} options.fragmentShader The fragment shader to use. The default <code>sampler2D</code> uniforms are <code>colorTexture</code> and <code>depthTexture</code>. The color texture is the output of rendering the scene or the previous post-process. The depth texture is the output from rendering the scene. The shader should contain one or both uniforms. There is also a <code>vec2</code> varying named <code>v_textureCoordinates</code> that can be used to sample the textures.
+     * @param {Object} [options.uniformValues] An object whose properties will be used to set the shaders uniforms. The properties can be constant values or a function. A constant value can also be a URI, data URI, or HTML element to use as a texture.
+     * @param {Number} [options.textureScale=1.0] A number in the range (0.0, 1.0] used to scale the texture dimensions. A scale of 1.0 will render this post-process to a texture the size of the viewport.
+     * @param {Boolean} [options.forcePowerOfTwo=false] Whether or not to force the texture dimensions to be both equal powers of two. The power of two will be the next power of two of the minimum of the dimensions.
```

Comment:
Perhaps I'll see it later...where is this used?

Reply:
This and `textureScale` are only currently used by the sun. Other effects could use it like bloom and AO for example.

Reply:
If you are confident it is generally useful then keep it as is; otherwise, perhaps consider making this private/undocumented for now.